EdFi.FiF.Etl/src/surveyETL.js

```diff
@@ -0,0 +1,184 @@
+    await client.connect()
+        .catch(e => console.error(e));
+    return client;
+}
+
+async function Extract(fileName) {
+    let questions = null;
+    let answers = [];
+    return new Promise(resolve => {
+        fs.createReadStream(fileName)
+            .pipe(csv())
+            .on('headers', headers => questions = headers)
+            .on('data', row => answers.push(row))
+            .on('end', () => {
+                resolve({ questions: questions, answers: answers });
+            });
+    });
+
+}
+
+async function Load(surveytitle, questions, answers, db) {
+
+    async function getOrSaveSurvey(surveytitle, db) {
+        return await db
+            .query("SELECT surveykey, title FROM fif.survey where title = $1;", [surveytitle])
+            .then(async result => {
+                if (result.rows.length == 0) {
+                    return await db
+                        .query("INSERT INTO fif.survey (title) VALUES($1) RETURNING surveykey, title;", [surveytitle])
+                        .then(result => result.rows[0])
+                }
+                return result.rows[0];
+            })
+            .catch(e => console.error(e))
+    }
+
+    async function getQuestions(surveykey, db) {
+        return await db.query("SELECT * FROM fif.surveyquestion sq where sq.surveykey = $1", [surveykey])
+            .then(result => result.rows);
+    }
+
+    async function getOrSaveQuestions(questions, surveykey, db) {
+        let dbQuestions = await getQuestions(surveykey, db);
+        let toSave = questions
+            .filter(h =>
+                !dbQuestions.some(q => q.question.toUpperCase() == h.toUpperCase()) &&
+                !METADATA_FIELDS.some(f => f.toUpperCase() == h.toUpperCase())
+            );
+        for (q in toSave) {
+            await db.query("INSERT INTO fif.surveyquestion (surveykey, question) VALUES($1, $2);", [surveykey, toSave[q]]);
+        }
+
+        return await db.query("SELECT * FROM fif.surveyquestion sq where sq.surveykey = $1", [survey.surveykey])
+            .then(result => result.rows);
+    }
+
+    async function getOrSaveStudentSurvey(surveykey, studentAnswers, db) {
+        let date = studentAnswers[SURVEY_DATE_FIELD];
+        let studentschoolkey = studentAnswers[STUDENT_SCHOOL_KEY_FIELD];
+
+        return await db
+            .query("SELECT studentsurveykey, surveykey, studentschoolkey, \"date\" FROM fif.studentsurvey where surveykey = $1 and studentschoolkey = $2; ", [surveykey, studentschoolkey])
+            .then(async result => {
+                if (result.rows.length > 0) { return result.rows[0] }
+                return await db
+                    .query("INSERT INTO fif.studentsurvey (surveykey, studentschoolkey, \"date\") VALUES($1, $2, $3) RETURNING *;", [surveykey, studentschoolkey, date])
+                    .then(result => result.rows[0])
+                    .catch(err => { console.error("ERROR: " + err.detail); return null; });
+            });
```

Comment:
```suggestion
        const result = await db
            .query("SELECT studentsurveykey, surveykey, studentschoolkey, \"date\" FROM fif.studentsurvey where surveykey = $1 and studentschoolkey = $2; ", [surveykey, studentschoolkey])
            
          if (result.rows.length > 0) { return result.rows[0] }
          return await db
                    .query("INSERT INTO fif.studentsurvey (surveykey, studentschoolkey, \"date\") VALUES($1, $2, $3) RETURNING *;", [surveykey, studentschoolkey, date])
                    .then(result => result.rows[0])
                    .catch(err => { console.error("ERROR: " + err.detail); return null; });
            });
```
I think we should put the ".then" into a separated logic, so with less indentation, we improve readability.

Reply:
done